crates/wasmtime/src/trampoline/create_handle.rs

```diff
@@ -26,17 +26,24 @@ pub(crate) fn create_handle(
     let module = Arc::new(module);
 
     unsafe {
-        let handle = InstanceHandle::new(
-            module,
-            &finished_functions,
-            imports,
-            store.memory_creator(),
-            &|_| shared_signature_id.unwrap(),
-            state,
-            store.interrupts(),
-            store.externref_activations_table() as *const VMExternRefActivationsTable as *mut _,
-            store.stack_map_registry() as *const StackMapRegistry as *mut _,
-        )?;
-        Ok(store.add_instance(handle))
+        // Use the default allocator when creating handles associated with host objects
+        let handle = store
+            .engine()
+            .config()
+            .default_instance_allocator
```

Comment:
As commented in a previous response, I viewed the "pooling instance allocator" as relating to *module* instantiations and it's not used for creating host objects because I didn't want them to count towards the allocator's limits (from the runtime's perspective, a host object instance and a module instance is basically indistinguishable).

Personally, I think this part of the design (specifically the notion of the "default instance allocator") is a bit wonky and I'd be happy to revisit it.

What I don't want, however, is users to define a `Wasi` instance and, unbeknownst to them, the availability in the pool has been reduced by many dozens of instances, especially when none of those instances defined any memories or tables (and thus the pool's memory is underutilized).  I view the way we instantiate host objects to be an implementation detail of Wasmtime and not one I really want to expose users to when thinking about "instances".

Doing it this way also means host objects are not restricted by any limits placed on the pooling allocator.

That said, I'm totally open to better ways of doing this.

Reply:
Ah ok that makes sense, and I agree that this seems like the best way to handle this. Agreed that creating a `Func` shouldn't accidentally reserve 6GB of address space because you're using a pooling allocator.

Could the comment here be expanded with some of this too?